backend/db/refresh.py

```diff
@@ -39,18 +39,26 @@ def refresh_db(
     local = use_local_db
     print('INFO: Starting database refresh.', flush=True)  # flush: for gh action
     t0, t0_str = datetime.now(), current_datetime()
-    update_db_status_var('last_refresh_request', t0_str, local)
+
+    if check_db_status_var('refresh_status') == 'active':
+        update_db_status_var('new_request_while_refreshing', t0_str, local)
+        return
+    
+    print('INFO: Starting database refresh.', flush=True)  # flush: for gh action
     update_db_status_var('refresh_status', 'active', local)
+    update_db_status_var('last_refresh_request', current_datetime(), local)
     try:
         with get_db_connection(local=local) as con:
             last_refresh = last_refresh_timestamp(con)
             if since and dp.parse(since) > dp.parse(last_refresh) and not force_non_contiguity:
                 raise ValueError(SINCE_ERR)
             since = since if since else last_refresh
 
-            # Refresh DB
+            # Refresh DB:
             # todo: will use instead when ready: all_new_objects_enclave_to_db()
-            csets_and_members_enclave_to_db(con, schema, since)
+            if not csets_and_members_enclave_to_db(con, schema, since):
+                print('INFO: Database is up to date, no refresh necessary')
+                return
 
```

Comment:
This was my idea for exiting out, I changed this function to return None when no refresh is necessary as suggested so it exits out. It should run as normal and update as expected if none is not returned.

Reply:
This change LGTM!